Vienna/Sources/Database/Database.m

```diff
@@ -1591,17 +1604,28 @@ -(BOOL)updateArticle:(Article *)existingArticle ofFolder:(NSInteger)folderID wit
     // Extract the data from the new state of article
     NSString * articleBody = article.body;
     NSString * articleTitle = article.title;
-    NSDate * articleDate = article.date;
     NSString * articleLink = article.link.vna_trimmed;
     NSString * userName = article.author.vna_trimmed;
     NSString * articleGuid = article.guid;
     NSInteger parentId = article.parentId;
     BOOL revised_flag = article.revised;
 
-    // Set some defaults
-    if (articleDate == nil) {
-        articleDate = existingArticle.date;
+    // keep last update date the same if not set in the current version of the article
+    NSDate * lastUpdate = article.lastUpdate && [article.lastUpdate isGreaterThan:existingArticle.lastUpdate]
+                            ? article.lastUpdate
+                            : existingArticle.lastUpdate;
+
+    // we never change the publication date, unless the date provided in the feed is prior to it and makes sense
+    NSDate * publicationDate = existingArticle.publicationDate;
+    if (article.publicationDate && [article.publicationDate isLessThan:existingArticle.publicationDate]) {
+        publicationDate = article.publicationDate;
     }
```

Comment:
Where is the "and makes sense" part in this? What would happen if it doesn´t "make sense"?